### ISSUE TITLE
After panel port change fail2ban does not ban new port

### ISSUE THREAD

**Issue by Suro** [serghey-rodin/vesta#1541]
### Operating System (OS/VERSION):

Debian 9

### VestaCP Version:

0.9.8  20

### Installed Software (what you got with the installer):

nginx, php-fpm, iptables, fail2ban, vsftpd, exim, dovecot, spamassassin, clamav, roundcube, mysql, postgresql

### Steps to Reproduce:
Change Vesta panel port to something other than the default 8083

### Related Issues/Forum Threads:

https://forum.vestacp.com/viewtopic.php?f=10&t=16585

### Other Notes:

After having to change the default port the panel runs on I noticed that the iptable chain was not updated. At the moment failed login attempts trigger a ban but only on the original port leaving the panel still accessible. 

Port for fail2ban is set in v-add-firewall-chain . Editing the port there and reloading fail2ban fixes the issue for now. As others suggested in the forum it would be nice to be able to set the port on install or in the vesta.conf file.

I think a simple solution for now would be to block all ports on a panel ban. I will try to see about that as soon as I get a chance. Not sure if you can just change 8083 to 1-65535 in v-add-firewall-chain?

**Comment by darkworks** [comment]
to change vesta port 
in vesta firewall change 8083 to your new desire port lets say 8000
then edit : `nano /usr/local/vesta/nginx/conf/nginx.conf`
```
# Vhost
    server {
        listen          8083;   #  change it to 8000
```
save and restart vesta like  
`/etc/init.d/vesta restart`

**Comment by nsuro** [comment]
Yes that is how the panel port is changed but the issue is that fail2ban does not work after that change.

**Comment by nsuro** [comment]
I tested changing this line in v-add-firewall-chain from
`    VESTA)      port=8083; protocol=TCP  ;;`
To
`    VESTA)      port='1:65535'; protocol=TCP  ;;`

This blocked access on all ports for my IP instead of just 8083. Maybe not the best way to handle it as I was locked out of every thing. Would either have to wait for ban to expire or login through console to unban.

Not sure if this is how you guys would like to address the issue, but just a suggestion :smiley: Thanks for all your hardwork!!

**Comment by coldplug** [comment]
Hello,

In order for fail2ban to ban on correct port, you need to change this file:
/usr/local/vesta/data/firewall/chains.conf

It is not documented well (if anywhere) and vestacp forum is full of this issue and there is no single answer to that even from vestacp stuff - looks like they don't know this setting either.

Restart firewall after change. Good luck

**Comment by nsuro** [comment]
That looks like it would be best since it might not get wiped after upgrade then. But it doesn't seem to work exactly though on debian 9.

- Change the port in chains.conf
- Run v-update-firewall to get iptables to update (restarting fail2ban or vesta didn't update)
- Reboot and found new port was still there
- Check chains.conf and found was back to original after reboot
- Now anytime v-update-firewall is run it sets it back to the original.

I'll try on fresh installs to see if its just messed something up ;)

**Comment by coldplug** [comment]
Ah, I see. Something changes the file on boot. I might look at it when I get time. In a meantime, put sed replace in /etc/rc.local or whatever Debian executes at the end of boot process:

sed -i "/SSH/c\CHAIN='SSH' PORT='2222' PROTOCOL='TCP'" /usr/local/vesta/data/firewall/chains.conf

On Centos this successfully reverted back the change.
